language/dsl/src/components.tsx

```diff
@@ -262,3 +262,13 @@ export function createSlot<SlotProps = unknown>(options: {
     );
   };
 }
+/** An object wrapper that tracks it's index from a parent */
+export const ObjectWithIndexTracking = (props: WithChildren) => {
```

Comment:
Can we add a test for this?

Reply:
What would testing something look like?

Reply:
Something similar to how we test the auto id functionality should be sufficient https://github.com/player-ui/tools/blob/main/language/dsl/src/__tests__/asset-api.test.tsx